### ISSUE TITLE
Filter via $filters for attributes doesn't work for frontend sdk

### ISSUE THREAD

**Issue by gauravkrp** [swellstores/swell-js#58]
I've been messing my brain around filtering products on client side by using `swell-js` sdk but it doesn't work.

````
const productsList = await swell.products.list({
    $filters: {
       featured_products: 'yes',
    },
  });
````

Somehow a different workaround that I found over internet is working - 
````
const productsList = await swell.products.list({
    'attributes.featured_products': 'yes',
});
````

Is it a bug or is the docs out of sync for front-end APIs ?
Swell version - `"swell-js": "^3.17.4",`

**Comment by stafyniaksacha** [comment]
Hello @gauravkrp 
I got same version while upgrading `swell-js` ....

You should stay on `3.14.0` (which matches master branch), also work on `3.15.0` but all others' version are broken

From what I understood they migrated the codebase from cjs to ejs, but the cjs version is no more available in the dist, so it fallback to umd build

**Comment by logeshswell** [comment]
@gauravkrp cc: @stafyniaksacha 

For the below payload, 
```
const productsList = await swell.products.list({
  $filters: {   // depth=1
     featured_products: 'yes',   // depth=2
  },
});
```

I noticed that `$filters` field wasn't converted into json properly in the Swell API request and identified that in `v3.16.0` upgrade, we missed the below regression that caused this issue. 

So, I guess the default depth value is taken as 1 for this case.

<img width="397" alt="Screenshot 2022-09-03 at 4 34 40 PM" src="https://user-images.githubusercontent.com/100766131/188268462-7efa7f42-697b-467f-af09-142ba6b65bdc.png">

[Diff link](https://github.com/swellstores/swell-js/compare/v3.15.0...v3.16.0) (`src/utils/index.js:103`)

We will fix this bug at the earliest and then close this issue. Thanks a lot @gauravkrp for bringing this up to us.

**Comment by gauravkrp** [comment]
@logeshswell One quick question Logesh. While I was struggling with how to filter products on the frontend using this `swell-js` library, I even tested to use REST APIs directly by calling `https://api.swell.store` with a Basic Auth in headers using store-id and secret-key. While REST API calls work when made via POSTMAN or a backend such as node.js, I'm getting a CORS error on my react.js frontend. 
Is there a way to solve this CORS issue on front-end or REST APIs don't work on frontend ??

**Comment by logeshswell** [comment]
@gauravkrp Hope this answers your question. If not, please reply to this thread and I am happy to answer you.
https://community.swell.is/discussion/638/can-i-use-swell-apis-directly-on-frontend-im-getting-cors-error

**Comment by logeshswell** [comment]
@gauravkrp cc: @stafyniaksacha This issue is fixed in `v3.17.6`
